### PR TITLE
refactor(www): Update react-icons from 2.2.7 to 3.x.x (latest)

### PR DESCRIPTION
**Pull request by mohouyizme**
## Related Issues
closes https://github.com/gatsbyjs/gatsby/issues/21859
closes https://github.com/gatsbyjs/gatsby/issues/21661

### CODE REVIEW

File: www/src/components/navigation.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-import TwitterIcon from "react-icons/lib/fa/twitter"
+import { GoMarkGithub as GithubIcon } from "react-icons/go"
+import {
+  FaTwitter as TwitterIcon,
+  FaDiscord as DiscordIcon,
+} from "react-icons/fa"
 
 import Link from "./localized-link"
 import SearchForm from "./search-form"
-import DiscordIcon from "./discord"
```
Comment (pieh) [inline www/src/components/navigation.js]: Can we delete https://github.com/gatsbyjs/gatsby/blob/master/www/src/components/discord.js file completely now? I think so

Comment (mohouyizme) [reply www/src/components/navigation.js]: Yes! it's useless.

Comment (pieh) [reply www/src/components/navigation.js]: I will delete it then after I finish my testing - making sure new import style as not impacting bundle sizes (I understand that the change is needed when switching to `react-icons@3`, just need to make sure we suddenly won't end up in full `react-icons` package being bundled)

**Comment by pieh** [comment]
Some icons also were relying on `vertial-align: middle` that was there in `react-icons@2` ( https://www.npmjs.com/package/react-icons#adjustment-css ), so I'll also add the `IconContext.Provider` to restore that (and likely remove all inline styles for `verticalAlign`)

Here's example from this PR:
Screenshot 2020-03-10 at 12 20 11
For comparison here's master:
Screenshot 2020-03-10 at 12 21 40

**Comment by mohouyizme** [comment]
I didn't notice that, thanks for the remark!

File: www/gatsby-config.js [hunk 2]
```diff
@@ -355,6 +355,7 @@ module.exports = {
         nodeTypes: [`StartersYaml`],
       },
     },
+    `@wardpeet/gatsby-plugin-sourcemap-explorer`,
```
Comment (pieh) [inline www/gatsby-config.js]: this is just temporary - will remove it later on

**Comment by fk** [comment]
Yay! Thank you @mohouyizme and @pieh! 🙏 🤗

**Pull request merged**